### PR TITLE
protractor_web_test & protractor_web_test_suite

### PR DESCRIPTION
**Pull request by gregmagolan**
Replaces #24370

### CODE REVIEW

File: integration/bazel/WORKSPACE [hunk 1]
```diff
@@ -6,16 +6,25 @@ workspace(name = "bazel_integration_test")
 
 http_archive(
     name = "build_bazel_rules_nodejs",
-    url = "https://github.com/bazelbuild/rules_nodejs/archive/0.9.1.zip",
-    strip_prefix = "rules_nodejs-0.9.1",
-    sha256 = "6139762b62b37c1fd171d7f22aa39566cb7dc2916f0f801d505a9aaf118c117f",
+    url = "https://github.com/gregmagolan/rules_nodejs/archive/bba77118cb18c7fdefa9cd4612d0477ab045a2c4.zip",
```
Comment (alexeagle) [inline integration/bazel/WORKSPACE]: rules_nodejs is released now, should have what you need?

File: WORKSPACE [hunk 2]
```diff
@@ -11,11 +11,20 @@ http_archive(
     sha256 = "2f77623311da8b5009b1c7eade12de8e15fa3cd2adf9dfcc9f87cb2082b2211f",
 )
 
+http_archive(
```
Comment (alexeagle) [inline WORKSPACE]: comment why we need this?

File: integration/bazel/WORKSPACE [hunk 3]
```diff
@@ -6,16 +6,25 @@ workspace(name = "bazel_integration_test")
...
+    ],
+    sha256 = "ddedc7aaeb61f2654d7d7d4fd7940052ea992ccdb031b8f9797ed143ac7e8d43",
 )
 
 http_archive(
     name = "io_bazel_rules_webtesting",
-    url = "https://github.com/bazelbuild/rules_webtesting/archive/v0.2.0.zip",
-    strip_prefix = "rules_webtesting-0.2.0",
-    sha256 = "cecc12f07e95740750a40d38e8b14b76fefa1551bef9332cb432d564d693723c",
+    url = "https://github.com/bazelbuild/rules_webtesting/archive/8fd9ce0fd9254bde251da0bc373d6cd08e811434.zip",
```
Comment (alexeagle) [inline integration/bazel/WORKSPACE]: should we ask them for a release?

File: integration/bazel/src/BUILD.bazel [hunk 4]
```diff
@@ -46,15 +48,11 @@ genrule(
...
-        "-p",
-        "5432",
-    ],
     data = [
         "index.html",
         ":bundle",
         ":zone.js",
     ],
     entry_point = "http-server/bin/http-server",
+    templated_args = ["src"],
```
Comment (alexeagle) [inline integration/bazel/src/BUILD.bazel]: don't I still need to specify a port for when I run the prodserver manually? what if I had two http-server targets and wanted to run both? would I give extra args on the `bazel run` command line instead?

File: integration/bazel/test/e2e/on-prepare.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import { browser } from 'protractor';
+import {OnPrepareConfig, runServer} from '@angular/bazel/protractor-utils';
+
+export = function(config: OnPrepareConfig) {
```
Comment (alexeagle) [inline integration/bazel/test/e2e/on-prepare.ts]: we'll need some docs explaining what bits you copy into your app, could be separate PR

File: integration/bazel/test/e2e/on-prepare.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+  const portFlag = config.server.endsWith('prodserver') ? '-p' : '-port';
```
Comment (alexeagle) [inline integration/bazel/test/e2e/on-prepare.ts]: comment why there are different flags

File: packages/bazel/src/protractor/protractor_web_test.bzl [hunk 7]
```diff
@@ -0,0 +1,341 @@
...
+      local=local,
+      shard_count=shard_count,
+      size=size,
+      tags=wrapped_test_tags,
+      timeout=timeout,
+      visibility=["//visibility:private"],
+      **remaining_keyword_args)
+
+  web_test_suite(
+      name=name,
```
Comment (alexeagle) [inline packages/bazel/src/protractor/protractor_web_test.bzl]: ugh the duplication is pretty gross :(

**Review by alexeagle — APPROVED** [review]

File: integration/bazel/test/e2e/on-prepare.ts (hunk 6 above)
Comment (gregmagolan) [reply integration/bazel/test/e2e/on-prepare.ts]: depending on if ts_devserver is running or http-server

File: integration/bazel/src/BUILD.bazel (hunk 4 above)
Comment (gregmagolan) [reply integration/bazel/src/BUILD.bazel]: yup. you could pass command line arguments. without them then you'd get the default port for prodserver.

**Comment by mhevery** [comment]
http://test/OCL:204156784:BASE:204156825:1531333178465:6bcc1513
